src/engine/engine_wt.c

```diff
@@ -18,6 +18,84 @@
 #define OCF_ENGINE_DEBUG_IO_NAME "wt"
 #include "engine_debug.h"
 
+static void _ocf_write_wt_update_bits(struct ocf_request *req)
+{
+	bool miss = ocf_engine_is_miss(req);
+	bool dirty_any = req->info.dirty_any;
+	bool repart = ocf_engine_needs_repart(req);
+
+	if (!miss && !dirty_any && !repart)
+		return;
+
+	ocf_hb_req_prot_lock_wr(req);
+
+	if (miss) {
+		/* Update valid status bits */
+		ocf_set_valid_map_info(req);
+	}
+
+	if (dirty_any) {
+		/* Writes goes to both cache and core, need to update
+		 * status bits from dirty to clean
+		 */
+		ocf_set_clean_map_info(req);
+	}
+
+	if (repart) {
+		OCF_DEBUG_RQ(req, "Re-Part");
+		/* Probably some cache lines are assigned into wrong
+		 * partition. Need to move it to new one
+		 */
+		ocf_user_part_move(req);
+	}
+
+	ocf_hb_req_prot_unlock_wr(req);
+}
+
+static void _ocf_write_wt_do_flush_metadata_compl(struct ocf_request *req,
+		int error)
+{
+	if (error)
+		req->error = error;
+
+	if (env_atomic_dec_return(&req->req_remaining))
+		return;
+
+	if (req->error)
+		ocf_engine_error(req, true, "Failed to write data to cache");
+
+	ocf_req_unlock_wr(ocf_cache_line_concurrency(req->cache), req);
+
+	req->complete(req, req->info.core_error ? req->error : 0);
+
+	ocf_req_put(req);
+}
+
+static int ocf_write_wt_do_flush_metadata(struct ocf_request *req)
+{
+	struct ocf_cache *cache = req->cache;
+
+	env_atomic_set(&req->req_remaining, 1); /* One core IO */
```

Comment:
This comment is misleading. This is actually refcount guard, it does not represent any IO.